internal/pkg/aws/codestar/codestar.go

```diff
@@ -0,0 +1,49 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package codestar provides a client to make API requests to AWS CodeStar Connections.
+package codestar
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/codestarconnections"
+)
+
+type api interface {
+	GetConnection(input *codestarconnections.GetConnectionInput) (*codestarconnections.GetConnectionOutput, error)
+}
+
+// Connection represents a client to make requests to AWS CodeStarConnections.
+type Connection struct {
```

Comment:
nit: Should this be `Codestar` so that methods like `ListConnections` can be added in the future?

Reply:
Goland yelled at me for the struct having the same name as the package, but now that we have the `api` interface, it is fine with it. Yes, will change.